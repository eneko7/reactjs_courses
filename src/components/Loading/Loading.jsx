import React from 'react';
import style from './Loading.scss';

const Loader = () => (
  <div className={style.loading}>
    <div className={style.movies_grid_wrapper_loading} id="loading">
      <div className={style.movies_grid_wrapper_loading} />
      <div className={style.movies_grid_wrapper_loading} />
      <div className={style.movies_grid_wrapper_loading} />
    </div>
    <div className={style.movies_grid_wrapper_loading_text} id="loading_text">
      <span className={style.movies_grid_wrapper_loading_text_span}>LOADING</span>
    </div>
  </div>
);

export default Loader;
