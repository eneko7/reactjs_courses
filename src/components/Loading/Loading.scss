@import "../../assets/constants/constants.scss";

.movies_grid_wrapper_loading {
  display: flex;
  justify-content: space-between;
  width: 70px;
  margin: 20px auto 5px auto;

  .movies_grid_wrapper_loading {
    width: 13px;
    height: 13px;
    border-radius: 50%;
    background-color: $roseColor;
    animation-name: loading;
    animation-duration: 1s;
    animation-iteration-count: infinite;
  }

  .movies_grid_wrapper_loading:nth-child(2) {
    animation-delay: 0.2s;
  }

  .movies_grid_wrapper_loading:last-child {
    animation-delay: 0.4s;
  }
}

.movies_grid_wrapper_loading_text {
  display: block;
  margin: 5px auto 0 auto;
  padding-bottom: 50px;
  font-size: 20px;
  color: #969797;
  text-align: center;
  font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
}

@keyframes loading {
  30% { transform: scale(1.3, 1.3); }
  70% { transform: scale(0.9, 0.9); }
}

.loading {
  position: fixed;
  z-index: 10;
  bottom: 2%;
  left: 45%;
}

@media (max-width: $tablet) {
  .loading {
    left: 42%;
  }
}

@media (max-width: $phone) {
  .loading {
    left: 35%;
  }
}
