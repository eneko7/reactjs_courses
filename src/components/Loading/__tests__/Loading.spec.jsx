import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Loading from '../Loading';

const renderer = new ShallowRenderer();
renderer.render(<Loading />);
describe('Loading', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
