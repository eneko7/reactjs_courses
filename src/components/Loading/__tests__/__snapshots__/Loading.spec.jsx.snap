// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<div
  className="loading"
>
  <div
    className="movies_grid_wrapper_loading"
    id="loading"
  >
    <div
      className="movies_grid_wrapper_loading"
    />
    <div
      className="movies_grid_wrapper_loading"
    />
    <div
      className="movies_grid_wrapper_loading"
    />
  </div>
  <div
    className="movies_grid_wrapper_loading_text"
    id="loading_text"
  >
    <span
      className="movies_grid_wrapper_loading_text_span"
    >
      LOADING
    </span>
  </div>
</div>
`;
